jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/BaseGenerator.java

```diff
@@ -289,4 +294,19 @@ protected void addSchemaLabels(BuildConfiguration.BuildConfigurationBuilder buil
             buildBuilder.labels(labels);
         }
     }
-}
+
+    public void checkMissingArtifacts() {
+        try {
+            final File finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(getProject());
+            if(finalOutputArtifact == null) {
+                log.error("Final output artifact file was not detected. The project may have not been built. HINT: try to compile and package your application prior to running the container image build task.");
+            } else if (isArtifactSameAsPrevious(finalOutputArtifact, finalOutputArtifact.getParentFile().toPath(), log)) {
+                log.warn( "Final output artifact is the same as previous build. You might have forgotten to compile and package your application after making changes.");
+            }
+        } catch (IOException | NoSuchAlgorithmException e) {
+            log.warn("Failed to check if final output artifact is the same as previous build.", e);
+        } catch (NullPointerException e) {
```

Comment:
I don't think we should catch NPE anywhere.

Reply:
Yes, I added it there temporarily because the NPE cause createAssembly tests in JavaExecGeneratorTest to fail.
To fix this, we need to provide build directory, build filename, etc. in both these tests.
Without these properties, JKubeProjectUtil.getFinalOutputArtifact() throws NPE.
![image](https://github.com/eclipse/jkube/assets/35092323/04cfb32a-8a58-4b7d-917a-aab81c348bf1)